### PR TITLE
Feature/9 console query

### LINKED ISSUE #9: Console query for register and discard operations
Child issue of #3

### PR DESCRIPTION
**Pull request by h0n9**
- Supports parcel, request, usage queries

```
/* Commands (expected hierarchy)
 *
 * amocli |- query |- balance <address>
 *		   |
 *		   |- parcel <parcelID>
 *		   |- request --buyer <address> --target <parcelID>
 *		   |- usage --buyer <address> --target <parcelID>
 */
```

(+ Hotfix for signing transaction and saving keys)

resolves https://github.com/amolabs/amoabci/issues/9

### CODE REVIEW

File: cmd/amocli/cmd/query/parcel.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+var ParcelCmd = &cobra.Command{
+	Use:   "parcel <parcelID>",
+	Short: "Get parcel's extra informations",
+	Args:  cobra.MinimumNArgs(1),
+	RunE:  parcelFunc,
+}
+
+func parcelFunc(cmd *cobra.Command, args []string) error {
+	tmp := strings.TrimLeft(args[0], "0x")
```
Comment (Lbird) [inline cmd/amocli/cmd/query/parcel.go]: What is this for?

File: cmd/amocli/cmd/query/request.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	keyMap := make(map[string]common.HexBytes)
+
+	keyMap["buyer"] = buyerAddr
+	keyMap["target"] = targetHex
+
+	keyMapJSON, err := json.Marshal(keyMap)
+	if err != nil {
+		return err
+	}
```
Comment (Lbird) [inline cmd/amocli/cmd/query/request.go]: It is a non-trivial job to form a keyMap here. Recommend to move this piece of code to QueryRequest().

File: cmd/amocli/cmd/query/usage.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+	keyMap := make(map[string]common.HexBytes)
+
+	keyMap["buyer"] = buyerAddr
+	keyMap["target"] = targetHex
+
+	keyMapJSON, err := json.Marshal(keyMap)
+	if err != nil {
+		return err
+	}
```
Comment (Lbird) [inline cmd/amocli/cmd/query/usage.go]: See comment to requestFunc()

**Review by Lbird — CHANGES_REQUESTED** [review]

File: cmd/amocli/cmd/query/parcel.go (hunk 1 above)
Comment (h0n9) [reply cmd/amocli/cmd/query/parcel.go]: I thought parcelID might be formatted in hex string.

Comment (Lbird) [reply cmd/amocli/cmd/query/parcel.go]: This is the only place where `0x` prefix is used. For consistency, remove this prefix constraint.

**Review by Lbird — APPROVED** [review]

**Pull request merged**